### PR TITLE
Snowflake ALTER TABLE: Add multiple columns

### PR DESCRIPTION
**Pull request by erevear**
### Brief summary of the change made

Attempt at a fix for https://github.com/sqlfluff/sqlfluff/issues/2500

Moving the ALTER TABLE ADD COLUMN statement elements into a Delimited object in order to add more than one at a time.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -960,47 +960,53 @@ class AlterTableTableColumnActionSegment(BaseSegment):
...
-                            "AUTOINCREMENT",
-                            "IDENTITY",
-                        ),
-                        OneOf(
-                            # ( <start_num>, <step_num> )
-                            Bracketed(
-                                Ref("NumericLiteralSegment"),
-                                Ref("CommaSegment"),
-                                Ref("NumericLiteralSegment"),
+                Sequence(
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:963]: You need to run `black` on this file. Guessing the indentation means some lines are too long and need to be moved down. `black` should be able to do this for you automatically.

File: test/fixtures/dialects/snowflake/snowflake_select_union.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ file:
...
           select_clause_element:
             literal: '2'
-    - orderby_clause:
-      - keyword: ORDER
-      - keyword: BY
-      - literal: '1'
+        unparsable:
+        - keyword: ORDER
+        - keyword: BY
+        - raw: '1'
```
Comment (tunetheweb) [inline test/fixtures/dialects/snowflake/snowflake_select_union.yml:25]: This looks wrong. But I just tried your branch and this works. Think you need to rerun the generate script to correct this:

```
python test/generate_parse_fixture_yml.py
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: test/fixtures/dialects/snowflake/snowflake_select_union.yml (hunk 2 above)
Comment (erevear) [reply test/fixtures/dialects/snowflake/snowflake_select_union.yml]: Yes! I re-ran the generator and this was corrected, so the file is no longer in the change

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (erevear) [reply src/sqlfluff/dialects/dialect_snowflake.py:963]: Got it! I installed the pre commit hook and black reformatted the file through that, so it passes linting. Does that solve the issue here, or is there another step?

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_snowflake.py:963]: Looks good now.

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 3]
```diff
@@ -960,47 +960,54 @@ class AlterTableTableColumnActionSegment(BaseSegment):
...
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:963]: I don’t think you need this Sequence as it only includes one component (the Delimited). Can we remove it?

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 3 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_snowflake.py:964]: Have made this change (and another unnecessary `Sequence`) so we can include this in 0.10.0.

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**